ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -609,13 +610,29 @@ - (BOOL)allNonOptionalFormItemsHaveAnswers {
 }
 
 - (BOOL)continueButtonEnabled {
-    return ([self numberOfAnsweredFormItems] > 0
-            && [self allAnsweredFormItemsAreValid]
-            && [self allNonOptionalFormItemsHaveAnswers]);
+    BOOL enabled = ([self numberOfAnsweredFormItems] > 0
+                    && [self allAnsweredFormItemsAreValid]
+                    && [self allNonOptionalFormItemsHaveAnswers]);
+    if (self.isBeingReviewed) {
+        return enabled && ![self.savedAnswers isEqualToDictionary:self.originalAnswers];
+    } else {
+        return enabled;
+    }
+}
+
+- (BOOL)skipButtonEnabled {
+    BOOL enabled = self.formStep.optional;
+    if (self.isBeingReviewed) {
+        return self.readOnlyMode ? NO : enabled && !ORKIsAnswerEmpty(self.originalAnswers);
+    } else {
+        return enabled;
+    }
 }
 
+
 - (void)updateButtonStates {
     _continueSkipView.continueEnabled = [self continueButtonEnabled];
+    _continueSkipView.skipButton.enabled = [self skipButtonEnabled];
```

Comment:
Do we want to be consistent and create a skipEnabled property on the continueSkipView?